Sources/StreamChatSwiftUI/ChatThreadList/ChatThreadListItem.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import SwiftUI
+
+/// View for the thread list item.
+public struct ChatThreadListItem: View {
+    @Injected(\.utils) private var utils
+    @Injected(\.chatClient) private var chatClient
+
+    var thread: ChatThread
+
+    public init(
+        thread: ChatThread
+    ) {
+        self.thread = thread
+    }
+
+    public var body: some View {
+        ChatThreadListItemContentView(
+            channelNameText: channelNameText,
+            parentMessageText: parentMessageText,
+            unreadRepliesCount: unreadRepliesCount,
+            replyAuthorName: latestReplyAuthor?.name ?? "",
+            replyAuthorUrl: latestReplyAuthor?.imageURL,
+            replyAuthorIsOnline: latestReplyAuthor?.isOnline ?? false,
+            replyMessageText: replyMessageText,
+            replyTimestampText: replyTimestampText
+        )
+    }
+
+    var parentMessageText: String {
+        var parentMessageText: String
+        if thread.parentMessage.isDeleted {
+            parentMessageText = L10n.Message.deletedMessagePlaceholder
+        } else if let threadTitle = thread.title {
+            parentMessageText = threadTitle
+        } else {
+            let formatter = MessagePreviewFormatter()
+            parentMessageText =  formatter.formatContent(for: thread.parentMessage)
+        }
+        return L10n.Thread.Item.repliedTo(parentMessageText.trimmed)
+    }
+
+    var replyMessageText: String {
+        guard let latestReply = thread.latestReplies.last else {
+            return ""
+        }
+
+        if latestReply.isDeleted {
+            return L10n.Message.deletedMessagePlaceholder
+        }
+
+        let formatter = MessagePreviewFormatter()
```

Comment:
here we can also reuse one preview formatter.

Reply:
Done ✅